src/ansys/aedt/core/filtersolutions_core/ideal_response.py

```diff
@@ -354,8 +354,11 @@ def frequency_response(
             of the defined frequency ranges, and the second is a
             list of the requested parameters.
         """
-        self.graph_setup.minimum_frequency = minimum_frequency
-        self.graph_setup.maximum_frequency = maximum_frequency
+
+        if minimum_frequency is not None:
+            minimum_frequency = self.graph_setup.minimum_frequency
+        if maximum_frequency is not None:
+            maximum_frequency = self.graph_setup.maximum_frequency
```

Comment:
Is this backwards? Shouldn't we be assigning the new value to self.graph_setup.minimum_frequency when the new value is not none?

```suggestion
        if minimum_frequency is not None:
            self.graph_setup.minimum_frequency = minimum_frequency
        if maximum_frequency is not None:
            self.graph_setup.maximum_frequency = maximum_frequency
```

If this didn't cause a unit test to fail, we need to improve the frequency_response unit test.

Reply:
The unit tests are not failing. What error are you expecting to have the tests improved?

Reply:
@ramin4667 These expressions were backwards and therefore the plot extents were not being applied. If the unit test verifies the plot extents are properly set, it should have been failing before you applied this correction.

Reply:
@ramin4667 I realize I didn't unresolve this conversation before so I'm unresolving it now until we figure out if the unit tests need to be fixed/enhanced.